src/ConnectionAutomaton.cpp

```diff
@@ -262,14 +267,22 @@ void ConnectionAutomaton::onConnectionFrame(
           outputFrameOrEnqueue(
               Frame_RESUME_OK(streamState_->resumeTracker_->impliedPosition())
                   .serializeOut());
-          for (auto it : streamState_->streams_) {
-            const StreamId streamId = it.first;
+          for (auto it = streamState_->streams_.begin(); it != streamState_->streams_.end();) {
+            const StreamId streamId = (*it).first;
 
-            if (streamState_->resumeCache_->isPositionAvailable(
-                    frame.position_, streamId)) {
-              it.second->onCleanResume();
+            ErrorStream errorStream;
+
+            if (streamState_->resumeCache_->isPositionAvailable(frame.position_, streamId)) {
+                requestHandler_->handleCleanResume(streamId, errorStream);
             } else {
-              it.second->onDirtyResume();
+                requestHandler_->handleDirtyResume(streamId, errorStream);
+            }
+
+            if (errorStream) {
+                it = streamState_->streams_.erase(it);
```

Comment:
I believe you should be sending endStream signal to the AbstractAutomaton to make sure it will do a proper cleanup of subscribers/subscriptions.